kie-server-parent/kie-server-router/kie-server-router-proxy/src/test/java/org/kie/server/router/repository/FileRepositoryTest.java

```diff
@@ -213,4 +214,38 @@ public void onConfigurationReloaded() {
         repoWithWatcher.close();
         repoWithWatcher.clean();
     }
+
+    @Test
+    public void testFileCreation() throws Exception {
+
+        File repositoryDirectory = new File("target" + File.separator + UUID.randomUUID().toString());
+        Configuration configuration = new Configuration();
+        ConfigurationMarshaller marshaller = new ConfigurationMarshaller();
+        File configFile = new File(repositoryDirectory, "kie-server-router.json");
+
+        configuration.addContainerHost("container1", "http://localhost:8080/server");
+        configuration.addContainerHost("container2", "http://localhost:8180/server");
+        
+        configuration.addServerHost("server1", "http://localhost:8080/server");
+        configuration.addServerHost("server2", "http://localhost:8180/server");
+
+        ContainerInfo containerInfo = new ContainerInfo("test1.0", "test", "org.kie:test:1.0");
+        configuration.addContainerInfo(containerInfo);
+
+        if (configFile.exists()) {
+            try (FileReader reader = new FileReader(configFile)){
+                
+                configuration = marshaller.unmarshall(reader);
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        
+        ConfigFileWatcher configFileWatcher = new ConfigFileWatcher(configFile.getParentFile().getAbsolutePath(), marshaller, configuration);
+
+        assertNotNull(configFileWatcher);
```

Comment:
we need to test if the configuration file is present on fs.

Reply:
@spolti I'm not sure how to do this using this constructor but maybe I can validate using this. https://github.com/kiegroup/droolsjbpm-integration/blob/1753048d3d0419b7a06540eafc14a60fa1250a99/kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/FileRepository.java#L47  and so doing some check.